handler/cache/handler_helpers.go

```diff
@@ -88,7 +87,7 @@ func (h *reqHandler) getResponseHook() func(*utils.FlexibleResponseWriter) {
 
 		h.Logger.Debugf("[%p] Response is cacheable! Caching metadata and parts...", h.req)
 
-		if rw.Code == http.StatusOK {
+		if rw.Code == http.StatusOK || rw.Code == http.StatusPartialContent {
```

Comment:
This will overwrite the saved ObjectMetadata.Code with 206 if we make subsequent requests for parts of a file - probably not what we want.